### ISSUE TITLE
suggestion

### ISSUE THREAD

**Issue by luojin** [lingochamp/FileDownloader#137]
Hi @Jacksgong , I am confused why didn't check status when the task calling **start();**
as the reference below:
https://github.com/lingochamp/FileDownloader/blob/master/library/src/main/java/com/liulishuo/filedownloader/FileDownloader.java#L242

when the second time calling FileDownloadQueueSet.start() , it will always throw exception from:
https://github.com/lingochamp/FileDownloader/blob/master/library/src/main/java/com/liulishuo/filedownloader/BaseDownloadTask.java#L380

so, forgive me with poor english, please help, thx a lot.

**Comment by Jacksgong** [comment]
Thanks for your suggestion.

Because for FileDownloader, that is very high-level exception, so, FileDownloader wants to THROW EXCEPTION to exposure the problem.

FileDownloader doesn't allow to start the task with its old task object, the task object in FileDownloader is very instability when it is running in FileDownloader, and instead, FileDownloader suggests that create a new task object or queue.

---

For more make sense solution, I will open the interface to judge whether the task object is running in FileDownloader, and whether it can start safely. Maybe this solution can cover your problem in another way.

**Comment by luojin** [comment]
@Jacksgong
Thanks for your reply.

In my case, I want to download files parallel, but the task may append after the  FileDownloadQueueSet.start() has been called. 
Task A, Task B and Task C is running, Task D wants to join together.

How can I deal with it ?

**Comment by Jacksgong** [comment]
I think in your case there isn't a queue, You just start each Task alone for free, in FileDownloader, it would be in parallel as default. 

```
/**Task A:**/ FileDownloader.getImpl().create(url1).setListener(listener).start();
/** Task B:**/ FileDownloader.getImpl().create(url2).setListener(listener).start();
/** Task C:**/ FileDownloader.getImpl().create(url3).setListener(listener).start();
/** Task D:**/ FileDownloader.getImpl().create(url3).setListener(listener).start();
```

You even can Ref TasksManagerDemoActivity in the demo project, it also seems like a queue, but in the demo, we start each task alone.

**Comment by luojin** [comment]
@Jacksgong 
nice!
I can solve my case by your suggestion.
